src/txadmission.cpp

```diff
@@ -576,7 +587,25 @@ bool ParallelAcceptToMemoryPool(Snapshot &ss,
     // block; we don't want our mempool filled up with transactions that can't
     // be mined yet.
     if (!CheckFinalTx(tx, STANDARD_LOCKTIME_VERIFY_FLAGS, &ss))
-        return state.DoS(0, false, REJECT_NONSTANDARD, "non-final");
```

Comment:
right now, ParallelAcceptToMemoryPool either accepts to the memory pool or rejects with an error and lets the caller figure out what to do with it.  I think that that API contract should be preserved -- in other words, let's return the error from PATMP, and then let the caller decide to push it onto a different queue, based on IsChainSynced()

Reply:
we basically do , the only difference here is in one case we can return a DOS and in the case of a message which goes into the waitQ we just return false, but no DOS...we don't want to be potentially DOSing txns which are going to have processing delayed but appear to be nonfinal, at least for the time being.

Reply:
ok, I create a new reject code, REJECT_WAITING so we can preserve the current function, then I test for the absence of that reject code when we process the rejects.